File: 예림/boj/1260.cpp
#include <bits/stdc++.h>
using namespace std;

int N, M, V;
vector<vector<int>> graph;
bool visited[1001];

void dfs(int now) {
	visited[now] = true;
	cout << now << " ";

	for (int next : graph[now]) {
		if (!visited[next]) {
			dfs(next);
		}
	}
}

void bfs(int start) {
	queue<int> q;
	visited[start] = true;
	q.push(start);

	while (!q.empty()) {
		int now = q.front();
		cout << now << " ";
		q.pop();

		for (int next : graph[now]) {
			if (!visited[next]) {
				visited[next] = true;
				q.push(next);
			}
		}
	}
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	cin >> N >> M >> V;
	graph.resize(N + 1);
	for (int i = 0; i < M; i++) {
		int a, b;
		cin >> a >> b;
		graph[a].push_back(b);
		graph[b].push_back(a);
	}

	for (int i = 1; i <= N; i++) {
		sort(graph[i].begin(), graph[i].end());
	}


	dfs(V);
	cout << "\n";

	memset(visited, false, sizeof(visited));
	bfs(V);
	cout << "\n";

	return 0;
}

File: 예림/boj/1260.java
import java.util.*;

public class Main {
    
    static int N, M, V;
    static int a, b;
    static boolean[][] graph;
    static boolean[] visited;
    
    public static void main(String args[]) {
      Scanner sc = new Scanner(System.in);
      
      N = sc.nextInt();
      M = sc.nextInt();
      V = sc.nextInt();
      
      graph = new boolean[N + 1][N + 1];
      visited = new boolean[N + 1];
      
      for(int i = 0; i < M; i++) {
          a = sc.nextInt();
          b = sc.nextInt();
          graph[a][b] = true;
          graph[b][a] = true;
      }
      
      dfs(V);
      System.out.println();
      
      visited = new boolean[N + 1];
      bfs(V);
    }
    
    static void dfs(int cur) {
        visited[cur] = true;
        System.out.print(cur + " ");
        
        for(int next = 1; next <= N; next++) {
            if(graph[cur][next] && !visited[next]) {
                dfs(next);
            }
        }
    }
    
    static void bfs(int cur) {
        Queue<Integer> q = new LinkedList<>();
        
        q.offer(cur);
        visited[cur] = true;
        
        while(!q.isEmpty()) {
            int now = q.poll();
            System.out.print(now + " ");
            
            for(int next = 1; next <= N; next++) {
                if(graph[now][next] && !visited[next]) {
                    q.offer(next);
                    visited[next] = true;
                }
            }
        }
    }
}